### PR TITLE
fix(fossid-webapp): Don't create issue when the URL cannot be mapped

### PR DESCRIPTION
**Pull request by nnobelis**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: model/src/main/kotlin/utils/Extensions.kt [hunk 1]
```diff
@@ -143,6 +143,7 @@ enum class PurlType(private val value: String) {
     DEBIAN("deb"),
     DRUPAL("drupal"),
     GEM("gem"),
+    GENERIC("generic"),
```
Comment (sschuberth) [inline model/src/main/kotlin/utils/Extensions.kt:146]: Commit message: Please link directly to https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#generic

Comment (sschuberth) [reply model/src/main/kotlin/utils/Extensions.kt:146]: Also note that we've historically been using `a-name` (see further up in the `enum`) for more or less the same purpose. However, `a-name` is a [NexusIq / Sonatype specific thing](https://help.sonatype.com/iqserver/product-information/referencing-package-url-%28purl%29-and-component-identifiers#ReferencingPackageURL(purl)andComponentIdentifiers-a-name). I'll probably do a follow-up PR with docs to explain this.

File: scanner/src/main/kotlin/scanners/fossid/FossId.kt [hunk 2]
```diff
@@ -825,7 +825,7 @@ class FossId internal constructor(
                 val snippetProvenance = it.url?.let { url ->
                     ArtifactProvenance(RemoteArtifact(url, Hash.NONE))
                 } ?: UnknownProvenance
-                val purlType = it.url?.let { url -> urlToPackageType(url, issues)?.toString() } ?: "generic"
+                val purlType = it.url?.let { url -> urlToPackageType(url).toString() }
```
Comment (sschuberth) [inline scanner/src/main/kotlin/scanners/fossid/FossId.kt:828]: Commit message title: "Do not create *an* issue *if* ..."

File: scanner/src/main/kotlin/scanners/fossid/FossId.kt [hunk 3]
```diff
@@ -893,11 +892,10 @@ class FossId internal constructor(
...
             else -> {
-                issues += FossId.createAndLogIssue(
-                    source = "FossId",
-                    message = "Cannot determine PURL type for url '$url' and provider '$provider'."
-                )
-                null
+                logger.warn {
+                    "Cannot determine PURL type for url '$url' and provider '$provider'. Falling back to 'generic'."
+                }
+                PurlType.GENERIC
```
Comment (sschuberth) [inline scanner/src/main/kotlin/scanners/fossid/FossId.kt:898]: I propose

                PurlType.GENERIC.also {
                    logger.warn {
                        "Cannot determine PURL type for url '$url' and provider '$provider'. Falling back to '$it'."
                    }
                }


to not hard-code the string representation of `PurlType.GENERIC`.

**Review by sschuberth — CHANGES_REQUESTED** [review]

**Comment by sschuberth** [comment]
@nnobelis commit-lint fails. You need to use reference-syntax for URLs to work around the line length limitation, like

```
[1]: https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#generic
```

and of course refer to `[1]` from somewhere in the text.

File: scanner/src/main/kotlin/scanners/fossid/FossId.kt (hunk 3 above)
Comment (nnobelis) [reply scanner/src/main/kotlin/scanners/fossid/FossId.kt:898]: Sorry there has been a mix up and I fixed it in the wrong branch.

It has been pushed now:
https://github.com/oss-review-toolkit/ort/compare/5416fa09f20251e030b0cda9958537a90426d79b..0f39a48bd367abfc01442350cee78eae4088334d

Comment (nnobelis) [reply scanner/src/main/kotlin/scanners/fossid/FossId.kt:898]: Wait, it's not there :(

**Review by sschuberth — APPROVED** [review]

**Pull request merged**